samples/cpp/ngraph_function_creation_sample/main.cpp

```diff
@@ -249,156 +221,118 @@ std::shared_ptr<ov::Function> createNgraphFunction() {
+        const std::string weights_path{argv[1]};
+        const std::string device_name{argv[2]};
 
         // -------- Step 1. Initialize OpenVINO Runtime Core object --------
-        slog::info << "Loading OpenVINO runtime" << slog::endl;
         runtime::Core core;
 
         slog::info << "Device info: " << slog::endl;
-        std::cout << core.get_versions(FLAGS_d) << std::endl;
+        slog::info << core.get_versions(device_name) << slog::endl;
 
         // -------- Step 2. Create network using ov::Function --------
+        slog::info << "Create model from weights: " << weights_path << slog::endl;
+        std::shared_ptr<ov::Function> model = createNgraphFunction(weights_path);
+        printInputAndOutputsInfo(*model);
+
+        OPENVINO_ASSERT(model->inputs().size() == 1, "Incorrect number of inputs for LeNet");
+        OPENVINO_ASSERT(model->outputs().size() == 1, "Incorrect number of outputs for LeNet");
+
+        ov::Shape input_shape = model->input().get_shape();
+        OPENVINO_ASSERT(input_shape.size() == 4, "Incorrect input dimensions for LeNet");
 
-        auto model = createNgraphFunction();
+        const ov::Shape output_shape = model->output().get_shape();
+        OPENVINO_ASSERT(output_shape.size() == 2, "Incorrect output dimensions for LeNet");
+
+        const auto classCount = output_shape[1];
+        OPENVINO_ASSERT(classCount <= 10, "Incorrect number of output classes for LeNet model");
 
         // -------- Step 3. Apply preprocessing --------
         const Layout tensor_layout{"NHWC"};
 
         // apply preprocessing
         // clang-format off
-        model = ov::preprocess::PrePostProcessor(model)
+        model = PrePostProcessor(model).
             // 1) InputInfo() with no args assumes a model has a single input
-            .input(ov::preprocess::InputInfo()
+            input(InputInfo().
                 // 2) Set input tensor information:
                 // - precision of tensor is supposed to be 'u8'
                 // - layout of data is 'NHWC'
-                .tensor(ov::preprocess::InputTensorInfo()
-                    .set_layout(tensor_layout)
-                    .set_element_type(element::u8))
-                // 3) Here we suppose model has 'NCHW' layout for input
-                .network(ov::preprocess::InputNetworkInfo()
-                    .set_layout("NCHW")))
-        // 4) Once the build() method is called, the preprocessing steps
+                tensor(InputTensorInfo().
+                    set_layout(tensor_layout).
+                    set_element_type(element::u8)).
+                // 3) Adding explicit preprocessing steps:
+                // - convert u8 to f32
+                // - convert layout to 'NCHW' (from 'NHWC' specified above at tensor layout)
+                preprocess(PreProcessSteps().
+                    convert_element_type(ov::element::f32). // WA for CPU plugin
+                    convert_layout("NCHW")). // WA for CPU plugin
```

Comment:
copy paste? These is not resize in the sample, so don't need these WA code for CPU and as well other explicit preprocessing operations

Reply:
fixed